File: Lab 1/echo.py
import socket

s = socket.socket()         # Create a socket object
host = "localhost"	    # On local host this time
port = 8000                 # Correct Port Number Required

aim_message = "ferdia"
message = "GET /echo-server.php?message=" + aim_message + " HTTP/1.1\r\n\r\n"

s.connect((host, port))
s.send(message)

print message
print s.recv(256)
s.close

File: Lab 1/echo.hs
import Network
import System.Environment
import System.IO
import Network.HTTP
import Data.List.Split

main :: IO ()
main = do
    [port, message] <- getArgs
    tcpHandler <- connectTo "localhost" $ PortNumber $ toEnum $ read port
    hSetBuffering tcpHandler  LineBuffering
    hPutStr tcpHandler $ buildHTTPMessage message
    hGetContents tcpHandler >>= putStr . parseResponse 

buildHTTPMessage :: String -> String
buildHTTPMessage msg = "GET /echo-server.php?message=" ++ msg ++ " HTTP/1.1\r\n\r\n"

parseResponse :: String -> String
parseResponse httpMsg = "Response:\n" ++ last (splitOn "\r\n\r\n" httpMsg) ++ "\n"
